### ISSUE TITLE
[4] Incorrect integer value: 'Mr Hacker' for column 'created_by' at row 1

### ISSUE THREAD

**Issue by PhilETaylor** [joomla/joomla-cms#30969]
### Steps to reproduce the issue

Joomla 4 -> Add Article -> Save
Edit same article 
Edit HTML of the page with inspector tools

change the value of `jform_created_by_id ` field

`<input type="hidden" id="jform_created_by_id" name="jform[created_by]" value="Mr Hacker" class="field-user-input " data-onchange="">`

### Expected result

Validation before hitting the database with a "string" 

### Actual result

`Save failed with the following error: Incorrect integer value: 'Mr Hacker' for column 'created_by' at row 1`

<img width="958" alt="Screenshot 2020-10-07 at 02 06 13" src="https://user-images.githubusercontent.com/400092/95275916-b0f9f600-0841-11eb-9c79-f4b5f3f57efd.png">

### Additional comments

Do the same again, but this time use a non valid integer number like 999999 

`<input type="hidden" id="jform_created_by_id" name="jform[created_by]" value="999999" class="field-user-input " data-onchange="">`

Save.

NOTE THAT 999999 PERSISTS IN THE DATABASE!!!!!!!! This id is invalid, its not a user id of an existing user.

<img width="783" alt="Screenshot 2020-10-07 at 02 07 57" src="https://user-images.githubusercontent.com/400092/95276035-0209ea00-0842-11eb-9ce0-26e48e7018d1.png">

**Comment by brianteeman** [comment]
Like all validation in Joomla it is broken. I've banged my head against a brick wall for 18 months over this trying to convince people

**Comment by Quy** [comment]
Related #29855

**Comment by PhilETaylor** [comment]
> Related #29855

In so much as the user not found message only - but the real reason for opening this issue was not for that - it was to highlight the complete lack of validation that is allowing a string to get as far as an integer database column, unfiltered and un checked. I should also NOT be able to set a new user id, of a user that doesn't exist. 

Probably should have been an email to @joomla/security but I had already posted before I realised the true depth of this issue.
